core/storage/user/gae_models.py

```diff
@@ -1501,6 +1659,29 @@ def has_reference_to_user_id(cls, user_id):
         """
         return cls.query(cls.user_id == user_id).get() is not None
 
+    @staticmethod
+    def get_user_id_migration_policy():
+        """UserContributionScoringModel has ID that contains user id and one
+        other field that contains user ID.
+        """
+        return base_models.USER_ID_MIGRATION_POLICY.CUSTOM
+
+    @classmethod
+    def migrate_model(cls, old_user_id, new_user_id):
+        """Migrate model to use the new user ID in the id and user_id.
+
+        Args:
+            old_user_id: str. The old user ID.
+            new_user_id: str. The new user ID.
+        """
+        for model in cls.query(cls.user_id == old_user_id).fetch():
+            model_values = model.to_dict()
+            new_id = '%s.%s' % (model.id.split('.')[0], new_user_id)
```

Comment:
Consider doing a replacement instead. (If so, can this be unified with the generic methods?)